src/database/hooks/useLocalDatabaseHook.js

```diff
@@ -51,15 +47,16 @@ const useLocalDatabaseHook = (withMigration = false) => {
       return null;
     }
 
-    if (isEnteringApp && shouldDbMigrated) {
+    if (shouldDbMigrated) {
       setMigrationStatus(MIGRATION_STATUS.MIGRATING);
       if (withMigration) await LocalDatabaseMigration.migrateDb();
       setMigrationStatus(MIGRATION_STATUS.MIGRATED);
       scopedMigrationStatus = MIGRATION_STATUS.MIGRATED;
     }
 
-    if (isEnteringApp && scopedMigrationStatus === MIGRATION_STATUS.NOT_MIGRATED) {
-      await LocalDatabaseMigration.migrateDb();
+    if (scopedMigrationStatus === MIGRATION_STATUS.NOT_MIGRATED) {
+      setMigrationStatus(MIGRATION_STATUS.MIGRATING);
+      if (withMigration) await LocalDatabaseMigration.migrateDb();
```

Comment:
<img src="https://avatars.githubusercontent.com/in/347564?s=41" alt="Image description" width="20" height="20">   CodeRabbit

This new logic seems to be more straightforward than the previous one, but it also appears to be less restrictive. Previously, the migration would only happen if `isEnteringApp` was true. Now, it seems like the migration could happen regardless of whether the app is entering or not. If this is the intended behavior, then it looks good. However, if there are cases where you don't want the migration to occur, you might need to reintroduce some conditions here.


<!-- This is an auto-generated comment by OSS CodeRabbit -->